src/components/reservation/ReservationModalBody.tsx

```diff
@@ -177,19 +172,20 @@ export const ReservationModalBody = ({
       );
     }
 
-    if (reservablePidsFromAnotherLibrary?.length && patron) {
-      const { patronId, name, emailAddress, preferredPickupBranch } = patron;
+    if (
+      reservablePidsFromAnotherLibrary?.length &&
+      patron &&
```

Comment:
We have a function that is more descriptive/made for the purpose of checking if a user is logged in that is called somthing like: `userIsLoggedIn` I think we should use that instead and I think that should be the first check in this logic.

Reply:
I can only find `isAnonymous` ? But I still need the `patron` check? So, is it necessary?

Reply:
I see what you mean.
But actually I think we should use `isAnonymous` as a part of the `usePatronData()` request by setting enabled: 
```js
{
  enabled: !isAnonymous()
}
``` 

In that way we do not send the patron request if it is not needed.